File: auth-trigger-dynamodb/src/triggers/post_confirm_insert_dynamodb.ts
import { PostConfirmationTriggerEvent, Context, PostConfirmationTriggerHandler, Callback } from 'aws-lambda';
import * as dynamodb from 'aws-sdk/clients/dynamodb';

export const handler: PostConfirmationTriggerHandler = 
  async (event: PostConfirmationTriggerEvent, context: Context, callback: Callback) => {

    const { sub, email, given_name, family_name, preferred_username } = event.request.userAttributes;
    const tableName = process.env.testUsersTable!;
    const test_user = {
      user_id: sub,
      first_name: given_name,
      last_name: family_name,
      stage_name: preferred_username,
      email: email,
    };

    try {
      const docClient = new dynamodb.DocumentClient();

      await docClient.put({
        TableName: tableName,
        Item: test_user,
      })
      .promise();

      console.log(JSON.stringify(test_user));

      callback(null, event);
    } catch (err) {
      console.log(err);
      callback(err, null);
    }
  }

File: py-auth-trigger-dynamodb/src/triggers/post_confirm_insert_dynamodb.py
import os
import boto3

def handler(event, context):

    pool_user = event['request']['userAttributes']
    test_user = {}
    test_user['user_id'] = pool_user['sub']
    test_user['first_name'] = pool_user['given_name']
    test_user['last_name'] = pool_user['family_name']
    test_user['stage_name'] = pool_user['preferred_username']
    test_user['email'] = pool_user['email']
    
    dynamodb = boto3.resource('dynamodb')
    table = dynamodb.Table(os.environ['USERS_TABLE'])
    table.put_item(Item=test_user)

    return event